src/ndg/make.rs

```diff
@@ -33,12 +36,21 @@ impl<'a, T, D> GridCubicSmoothingSpline<'a, T, D>
         let permuted_axes: D = permute_axes(ndim);
 
         for ax in (0..ndim).rev() {
-            let x = breaks[ax].view();
-            let y = to_2d_simple(coeffs.view())?;
+            let x: ArrayView1<T> = breaks[ax].view();
+            let y: ArrayView2<T> = to_2d_simple(coeffs.view())?;
 
             let weights = self.weights[ax].map(|v| v.reborrow());
             let s = self.smooth[ax];
 
```

Comment:
@espdev , I found out that we need to add an additional trait binding in order to allow for sparse matrix addition (example in trait.rs at the bottom). However, once I add this, I get a recursion error that baffles me. It seems as if I a `CsMatBase` with itself as a type is created. I searched for a long time to no avail. Do you have an idea what could cause this compiler error?